File: 128_lcs.py
class Solution(object):
    def longestConsecutive(nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        record = set()
        for elem in nums:
            record.add(elem)
        res,curr = 1,1
        for elem in record:
            if elem - 1 not in record:
                while elem + 1 in record:
                    elem += 1
                    curr += 1
                res = max(curr,res)
                # update curr_val to initial
                curr = 1
            
        return res
nums = [100,4,200,1,3,2]
print(Solution.longestConsecutive(nums)) # 4

File: array/128_lcs.java
class Solution{
    public int longestConsecutive (int[] nums){
        // use hashSet
        Set<Integer> record = new HashSet<>();
        for (int num:nums){
            record.add(num);
        }
        // traverse and compare with curr_max
        int res = 0;
        for (int i=0;i < nums.length;i++){
            int elem = nums[i];
            int curr = 1;
            if (!(record.contains(elem-1))){
                while (record.contains(elem+1)){
                // continue moving
                curr += 1;
                elem += 1;
            }
        }
        res = Math.max(res,curr);
    }
        return res;
    }
}